source/crypto.c

```diff
@@ -56,7 +85,15 @@ PyObject *aws_py_md5_new(PyObject *self, PyObject *args) {
         return PyErr_AwsLastError();
     }
 
-    return PyCapsule_New(md5, s_capsule_name_hash, s_hash_destructor);
+    PyObject *capsule = PyCapsule_New(md5, s_capsule_name_hash, s_hash_destructor);
+
+    /* Why do we return NULL instead of PyErr */
```

Comment:
Remove this comment. Check the [doc](https://docs.python.org/3/c-api/capsule.html#c.PyCapsule_New) here.
> On failure, set an exception and return NULL.

Basically those functions will set an exception on failure.